### PR TITLE
Remove non OSS features and tests that checks any ES clusters != 7.X

### LINKED ISSUE #10: Remove tests, code changes that are not related oss
logstash-output-elasticsearch can use non oss features of ES. Since they won't be supported in logstash-output-opensearch, we need to remove this .

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Remove multiple versions
Remove groovy scripts
Remove license checker
clean up data stream
Delete xpack files
Update ADMINS
Remove iml files
Set github workflow
Use opensearch basic template
Clean up build and test files

### Issues Resolved
#10
#3 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
CI from my fork is succeeded: https://github.com/VijayanB/logstash-output-opensearch-1/actions/runs/948043352

File: lib/logstash/outputs/elasticsearch/ilm.rb [hunk 1]
```diff
@@ -1,116 +1,7 @@
 module LogStash; module Outputs; class ElasticSearch
   module Ilm
-
```
Comment (jmazanec15) [inline lib/logstash/outputs/elasticsearch/ilm.rb:3]: Cant this file be removed? I thought ilm is xpack

File: lib/logstash/plugin_mixins/elasticsearch/api_configs.rb [hunk 2]
```diff
@@ -11,15 +11,6 @@ module APIConfigs
         # Password to authenticate to a secure Elasticsearch cluster
         :password => { :validate => :password },
```
Comment (jmazanec15) [inline lib/logstash/plugin_mixins/elasticsearch/api_configs.rb:12]: Does this still work?

File: spec/fixtures/_nodes/7x.json [hunk 3]
```diff
@@ -20,7 +20,6 @@
       ],
       "attributes" : {
         "ml.machine_memory" : "17179869184",
```
Comment (jmazanec15) [inline spec/fixtures/_nodes/7x.json:22]: Cant we remove all ml related code as its xpack?

File: spec/integration/outputs/parent_spec.rb [hunk 4]
```diff
@@ -1,102 +1,101 @@
...
-                   "mapping.single_type" => true
-                 }})
-        end
-        Manticore.put("#{index_url}", {:body => mapping.to_json, :headers => default_headers}).call
-        pdoc = { "message" => "ohayo", join_field => parent_relation }
-        Manticore.put("#{index_url}/#{type}/#{parent_id}", {:body => pdoc.to_json, :headers => default_headers}).call
-
-        subject.register
-        subject.multi_receive(event_count.times.map { LogStash::Event.new("link_to" => parent_id, "message" => "Hello World!", join_field => child_relation) })
+      if ESHelper.es_version_satisfies?('<6')
```
Comment (jmazanec15) [inline spec/integration/outputs/parent_spec.rb:39]: Why do we need this?

**Comment by jmazanec15** [comment]
Can we remove?
1. logstash-output-opensearch/lib/logstash/outputs/elasticsearch/ilm.rb
2. logstash-output-opensearch/lib/logstash/outputs/elasticsearch/templates/ecs-disabled/

**Comment by jmazanec15** [comment]
Can this be removed? 
1. https://github.com/VijayanB/logstash-output-opensearch/blob/refactor-code/lib/logstash/outputs/elasticsearch.rb#L93

File: lib/logstash/plugin_mixins/elasticsearch/api_configs.rb (hunk 2 above)
Comment (VijayanB) [reply lib/logstash/plugin_mixins/elasticsearch/api_configs.rb:12]: I have ticket #5 check security . Will fix all security level clean up after having working CI for secure clusters.

File: spec/fixtures/_nodes/7x.json (hunk 3 above)
Comment (VijayanB) [reply spec/fixtures/_nodes/7x.json]: Ack

File: lib/logstash/outputs/elasticsearch/ilm.rb (hunk 1 above)
Comment (VijayanB) [reply lib/logstash/outputs/elasticsearch/ilm.rb:3]: Ack

File: spec/integration/outputs/parent_spec.rb (hunk 4 above)
Comment (VijayanB) [reply spec/integration/outputs/parent_spec.rb]: Removed

**Comment by VijayanB** [comment]
> Can we remove?
> 
> 1. logstash-output-opensearch/lib/logstash/outputs/elasticsearch/ilm.rb - done 
> 2. logstash-output-opensearch/lib/logstash/outputs/elasticsearch/templates/ecs-disabled/ -> This is open source. I had kept it for now.

**Review by dblock — APPROVED** [review]
Don't squash these changes!

**Review by jmazanec15 — APPROVED** [review]

**Pull request merged**